### PR TITLE
xbutil: read sc present state before issuing any SC related packets

### PR DESCRIPTION
**Pull request by rajkumar-xilinx**
CR-1054786

### CODE REVIEW

**Comment by maxzhen** [comment]
retest this please

File: src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp [hunk 1]
```diff
@@ -106,6 +106,11 @@ int XMC_Flasher::xclUpgradeFirmware(std::istream& tiTxtStream) {
     int retries = 5;
     int ret = 0;
 
+    if (!hasSC()) {
+        std::cout << "ERROR: SC is not present on platform" << std::endl;
+        return -EINVAL;
+    }
+
     if (!isXMCReady())
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp]: Have you tested on U50, U200?

The logic should be if no XMC, SC won't be there, so we should check XMC first.

You changed the logic to if has SC then check XMC, we might miss some XMC status.

File: src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp [hunk 2]
```diff
@@ -516,17 +524,23 @@ bool XMC_Flasher::isBMCReady()
...
     }
 
-    if (val) {
-        bool bmcReady = (BMC_MODE() == 0x1);
-        if (!bmcReady) {
-            xrt_core::ios_flags_restore format(std::cout);
-            std::cout << "ERROR: SC is not ready: 0x" << std::hex
+    if (val)
+        return true;
+    return false;
```
Comment (xdavidz) [inline src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp]: return (val != 0);

**Comment by chvamshi-xilinx** [comment]
retest this please

File: src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp (hunk 1 above)
Comment (rajkumar-xilinx) [reply src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp]: We have  ""if (!hasXMC()) return false;"" check in hasSC() function. So, we are checking XMC first and then checking sc_presence state.

Comment (xdavidz) [reply src/runtime_src/core/pcie/tools/xbmgmt/xmc.cpp]: ok

**Comment by maxzhen** [comment]
@rajkumar-xilinx is CR-1054786 approved for PU2 branch?

**Comment by chvamshi-xilinx** [comment]
> 
> 
> @rajkumar-xilinx is CR-1054786 approved for PU2 branch?

@maxzhen , No approval yet.  Sent an email to Sumit for the approval. Added you in the mail.

**Comment by maxzhen** [comment]
This fix has been approved by Sumit. Merging it now.

**Pull request merged**